src/controllers/userController.ts

```diff
@@ -0,0 +1,60 @@
+import { Request, Response } from 'express';
+import User from '../models'
+import bcrypt from 'bcrypt';
+import Joi from 'joi';
+
+// Joi schema for password validation
+const passwordSchema = Joi.string().min(8).required().messages({
+    'string.min': 'Password must be at least 8 characters long',
+    'any.required': 'Password is required'
+});
+
+//Function to check if the old passsword is valid and then changes it to the new one
+const updatePassword = async (req: Request, res: Response) => {
+
+    const { userId ,oldPassword, newPassword, confirmPassword } = req.body;
+
+    try {
+        // Find the user by ID
+        const user = await User.findByPk(userId);
+
+        if (!user) {
+            return res.status(404).json({ message: 'User not found' });
+        }
+
+        // Check if the provided old password matches the one stored in the database
+        const passwordMatch = await bcrypt.compare(oldPassword, user.password);
+        if (!passwordMatch) {
+            console.log(passwordMatch);
+            return res.status(401).json({ message: 'Invalid old password' });
+        }
+
+        if (newPassword !== confirmPassword) {
+            return res.status(401).json({ message: 'Password must match' });
+        }
+
+       // Validate the new password
+       const { error: passwordError } = passwordSchema.validate(newPassword);
+       if (passwordError) {
+            const errorMessage = passwordError.details[0].message;
+            return res.status(400).json({ message: errorMessage });
+        }
+
+        // Hash the new password
```

Comment:
Reduce the usage of comments. They're used in areas where it's hard to understand for your teammates